### PR TITLE
TST: Enable cache directories on AppVeyor.

### PR DESCRIPTION
**Pull request by QuLogic**

### CODE REVIEW

**Comment by QuLogic** [comment]
Grr, of course, in order to get the cache to help, we need to have one run actually work...

**Comment by dstansby** [comment]
I'll watch the Appveyor build and restart it until it passes...

**Comment by QuLogic** [comment]
If this finally works this time, I'm going to push another commit and hopefully we'll see that the cache is working.

**Comment by QuLogic** [comment]
Pushed another commit to hopefully fix caching on Python 2; 50% of AppVeyor builds are on Python 2.

File: setupext.py [hunk 1]
```diff
@@ -1146,8 +1146,11 @@ def do_custom_build(self):
...
-                        # fail on Lpy, oh well
-                        os.makedirs('build', exist_ok=True)
+                        os.makedirs('build')
+                    except OSError:
+                        # Don't care if it exists.
+                        pass
+                    try:
                         shutil.copy(tarball_cache_path, tarball_path)
                         print('Using cached tarball: {}'
                               .format(tarball_cache_path))
```
Comment (anntzer) [inline setupext.py]: The except just below could probably be just for OSError?

**Comment by QuLogic** [comment]
On Travis, [Python 2.7](https://travis-ci.org/matplotlib/matplotlib/jobs/208068186#L1510), [Python 3.4](https://travis-ci.org/matplotlib/matplotlib/jobs/208068188#L1434), [Python 3.4](https://travis-ci.org/matplotlib/matplotlib/jobs/208068191#L1442), [Python 3.5](https://travis-ci.org/matplotlib/matplotlib/jobs/208068192#L1444), [Python 3.6](https://travis-ci.org/matplotlib/matplotlib/jobs/208068193#L1756) and [Python nightly](https://travis-ci.org/matplotlib/matplotlib/jobs/208068194#L1437) are all using the cached tarball now.

AppVeyor [is not](https://ci.appveyor.com/project/matplotlib/matplotlib/build/1.0.561/job/q58od458878xcmgx#L494), still. I don't know what's up with it. Maybe caching only works on `master`?

File: setupext.py (hunk 1 above)
Comment (QuLogic) [inline setupext.py]: This is fixed, BTW.

**Comment by QuLogic** [comment]
When I run this on `master` in my own copy, AppVeyor includes some more [lines about updating the build cache](https://ci.appveyor.com/project/QuLogic/matplotlib/build/1.0.2/job/4x427b02e03e5kwb#L4042). These do not appear in the builds here, so I assume this won't be enabled until it's merged into `master`.

**Comment by QuLogic** [comment]
The second build shows [AppVeyor restoring the cache](https://ci.appveyor.com/project/QuLogic/matplotlib/build/1.0.3/job/ob36jxurfaw2y837#L6) and Matplotlib [using the cached tarball](https://ci.appveyor.com/project/QuLogic/matplotlib/build/1.0.3/job/ob36jxurfaw2y837#L495), so I think this needs to be merged to have any effect on AppVeyor.

Interestingly, for those builds, I commented out the test run and it turns out that installing dependencies and building Matplotlib (note, this occurs _twice_ on AppVeyor) take up about 50% of the test wall time, except in the last full-test case.

**Comment by anntzer** [comment]
Something like https://github.com/mozilla/sccache / https://github.com/inorton/cclash / https://github.com/frerich/clcache (basically ccache with msvc support) could be useful?

**Comment by QuLogic** [comment]
I was thinking of something like that, but never investigated it. Probably something for another time, when the build queue is not so long...

File: setupext.py (hunk 1 above)
Comment (NelleV) [inline setupext.py]: :+1:

**Comment by NelleV** [comment]
I think we can merge this PR. It looks good to me.

**Comment by dstansby** [comment]
Lets see what happens once appveyor gets round to this merge...

**Pull request merged**